src/platforms/hosted/jlink_protocol.h

```diff
@@ -25,52 +26,334 @@
+ * │ JLINK_CMD_TARGET_RELEASE_RESET_HALT_RETRY   │ §5.6.2 EMU_CMD_HW_RELEASE_RESET_STOP_EX    │
+ * │ JLINK_CMD_TARGET_RELEASE_RESET_HALT_TIMEOUT │ §5.6.3 EMU_CMD_HW_RELEASE_RESET_STOP_TIMED │
+ * │ JLINK_CMD_TARGET_GET_CPU_CAPABILITIES       │ §5.6.6 EMU_CMD_GET_CPU_CAPS                │
+ * │ JLINK_CMD_TARGET_EXECUTE_CPU_CMD            │ §5.6.7 EMU_CMD_EXEC_CPU_CMD                │
+ * │ JLINK_CMD_TARGET_WRITE_MEMORY               │    -   EMU_CMD_WRITE_MEM                   │
+ * │ JLINK_CMD_TARGET_READ_MEMORY                │    -   EMU_CMD_READ_MEM                    │
+ * │ JLINK_CMD_TARGET_WRITE_MEMORY_ARM79         │ §5.6.8 EMU_CMD_WRITE_MEM_ARM79             │
+ * │ JLINK_CMD_TARGET_READ_MEMORY_ARM79          │ §5.6.9 EMU_CMD_READ_MEM_ARM79              │
+ * │ JLINK_CMD_TARGET_MEASURE_RTCK_REACTION_TIME │ §5.4.4 EMU_CMD_MEASURE_RTCK_REACT          │
+ * │ JLINK_CMD_TARGET_GET_CONNECTION_STATE       │ §5.4.3 EMU_CMD_GET_COUNTERS                │
+ * └─────────────────────────────────────────────┴────────────────────────────────────────────┘
+ */
+#define JLINK_CMD_TARGET_RELEASE_RESET_HALT_RETRY   0xd0U /* Resets the CPU and halts ASAP (fails after n retries) */
+#define JLINK_CMD_TARGET_RELEASE_RESET_HALT_TIMEOUT 0xd1U /* Resets the CPU and halts ASAP (fails after timeout) */
+#define JLINK_CMD_TARGET_GET_CPU_CAPABILITIES       0xe9U /* Get the capabilities of the target CPU */
+#define JLINK_CMD_TARGET_EXECUTE_CPU_CMD            0xeaU /* Executes target CPU functions */
+#define JLINK_CMD_TARGET_WRITE_MEMORY               0xf4U /* Write to target memory */
+#define JLINK_CMD_TARGET_READ_MEMORY                0xf5U /* Read from target memory */
+#define JLINK_CMD_TARGET_WRITE_MEMORY_ARM79         0xf7U /* Write to target memory on ARM 7/9 targets */
+#define JLINK_CMD_TARGET_READ_MEMORY_ARM79          0xf8U /* Read from target memory on ARM 7/9 targets */
+#define JLINK_CMD_TARGET_MEASURE_RTCK_REACTION_TIME 0xf6U /* Measure RTCK reaction time of the target device */
+#define JLINK_CMD_TARGET_GET_CONNECTION_STATE       0xc2U /* Get target connection timer counters */
+
+/* Configuration commands
+ *
+ * ┌────────────────────────┬────────────────────────────────┐
+ * │  BMDA J-Link command   │ RM08001 J-Link USB Protocol RM │
+ * ├────────────────────────┼────────────────────────────────┤
+ * │ JLINK_CMD_CONFIG_READ  │ §5.7.1 EMU_CMD_READ_CONFIG     │
+ * │ JLINK_CMD_CONFIG_WRITE │ §5.7.2 EMU_CMD_WRITE_CONFIG    │
+ * └────────────────────────┴────────────────────────────────┘
+ */
+#define JLINK_CMD_CONFIG_READ  0xf2U /* Read the probe configuration */
+#define JLINK_CMD_CONFIG_WRITE 0xf3U /* Write the probe configuration */
+
+/* J-Link hardware version - JLINK_CMD_INFO_GET_HARDWARE_VERSION */
+#define JLINK_HARDWARE_VERSION_TYPE_JLINK    0U
+#define JLINK_HARDWARE_VERSION_TYPE_JTRACE   1U
+#define JLINK_HARDWARE_VERSION_TYPE_FLASHER  2U
+#define JLINK_HARDWARE_VERSION_TYPE_JLINKPRO 3U
```

Comment:
These seem to be duplicates of the below definitions and the below ones make a bit more sense in terms of order of definitions. Please remove these.

Reply:
Yep, unintended dupe